mkosi

```diff
@@ -937,6 +937,10 @@ def reset_machine_id(args, workspace, run_build_script):
         machine_id = os.path.join(workspace, 'root', 'etc/machine-id')
         os.unlink(machine_id)
         open(machine_id, "w+b").close()
+        dbus_machine_id = os.path.join(workspace, 'root', 'var/lib/dbus/machine-id')
+        if os.path.exists(dbus_machine_id):
+            os.unlink(dbus_machine_id)
+            os.symlink('../../../etc/machine-id', dbus_machine_id)
```

Comment:
true, it's pretty safe to assume that. Whenever I see something like this it immediately raises my eyebrows though, and I'd always avoid any raised eyebrows for this, given how simply it is.

And yes, I think it makes sense not to create the symlink if the file doesn't exist anyway. But that's orthogonal to whether we unlink first or not:

```python
unlinked = False
try:
    os.unlink(dbus_machine_id)
    unlinked = True
except FileNotFoundError:
    pass

if unlinked:
    os.symlink(....)
``` 
or something along those lines